Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -66,38 +42,74 @@ import BraintreeCore
-
-    /// Optional. UI Customization for 3DS2 challenge views.
-    public var v2UICustomization: BTThreeDSecureV2UICustomization?
-
-    /// Optional. Sets all UI types that the device supports for displaying specific challenge user interfaces in the 3D Secure challenge.
-    ///
-    /// Defaults to `.both`
-    public var uiType: BTThreeDSecureUIType = .both
-
-    /// Optional. List of all the render types that the device supports for displaying specific challenge user interfaces within the 3D Secure challenge.
-    ///
-    /// - Note: When using `BTThreeDSecureUIType.both` or `BTThreeDSecureUIType.html`, all `BTThreeDSecureRenderType` options must be set.
-    /// When using `BTThreeDSecureUIType.native`, all `BTThreeDSecureRenderType` options except `.html` must be set.
-    public var renderTypes: [BTThreeDSecureRenderType]?
-
+    
     /// A delegate for receiving information about the ThreeDSecure payment flow.
     public weak var threeDSecureRequestDelegate: BTThreeDSecureRequestDelegate?
     
-    // MARK: - Internal Properties
+    // MARK: - Initializer
     
-    /// The dfReferenceID for the session. Exposed for testing.
-    var dfReferenceID: String?
+    /// Creates a `BTThreeDSecureRequest`
+    /// - Parameters:
+    ///    - accountType: Optional. The account type selected by the cardholder. Some cards can be processed using either a credit or debit account and cardholders have the option to choose which account to use.
+    ///    - additionalInformation: Optional. The additional information used for verification.
+    ///    - amount: The amount for the transaction.
+    ///    - billingAddress: Optional. The billing address used for verification
+    ///    - cardAddChallengeRequested: Optional.  An authentication created using this flag should only be used for vaulting operations (creation of customers' credit cards or payment methods) and not for creating transactions. If set to `true`, a card-add challenge will be requested from the issuer. If set to `false`, a card-add challenge will not be requested. If the parameter is missing, a card-add challenge will only be requested for $0 amount.
+    ///    - challengeRequested: Optional. If set to true, an authentication challenge will be forced if possible.
+    ///    - customFields: Object where each key is the name of a custom field which has been configured in the Control Panel. In the Control Panel you can configure 3D Secure Rules which trigger on certain values.
+    ///    - dataOnlyRequested:  Optional. Indicates whether to use the data only flow. In this flow, frictionless 3DS is ensured for Mastercard cardholders as the card scheme provides a risk score for the issuer to determine whether to approve. If data only is not supported by the processor, a validation error will be raised. Non-Mastercard cardholders will fallback to a normal 3DS flow.
+    ///    - dfReferenceID: The dfReferenceID for the session. Exposed for testing.
```

Comment:
Should this be exposed for merchants? It looks like it was internal previously.

Reply:
Ah, I see it looks like this is exposed for merchants using `prepareLookup`: https://github.com/braintree/braintree_ios/blob/5ecb18dc594019e5bf9292fd67871b8f15812fd1/Sources/BraintreeThreeDSecure/BTThreeDSecureClient.swift#L126-L128. We should update this docstring or have 2 inits for this. What do you think?

Reply:
ahhh that's a good point! i've updated the docstrings to say, that the `dfReferenceID` is helpful for merchants using the 3DS `prepareLookup` in b6f5382aefb7dc7a3467a891735988bba0a80a0d